api/src/paths/spatial/regions.ts

```diff
@@ -0,0 +1,147 @@
+  tags: ['spatial', 'regions'],
+  security: [
+    {
+      Bearer: []
+    }
+  ],
+  requestBody: {
+    content: {
+      'application/json': {
+        schema: {
+          properties: {
+            features: {
+              description: 'One or more GeoJSON features',
+              type: 'array',
+              items: GeoJSONFeature
+            }
+          }
+        }
+      }
+    }
+  },
+  responses: {
+    200: {
+      description: '200 Ok',
+      content: {
+        'application/json': {
+          schema: {
+            type: 'object',
+            required: ['regions'],
+            properties: {
+              regions: {
+                type: 'array',
+                items: {
+                  type: 'object',
+                  required: ['regionName', 'sourceLayer'],
+                  properties: {
+                    regionName: {
+                      type: 'string'
+                    },
+                    sourceLayer: {
+                      type: 'string'
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    400: {
+      $ref: '#/components/responses/400'
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    403: {
+      $ref: '#/components/responses/401'
+    },
+    500: {
+      $ref: '#/components/responses/500'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+/**
+ * Given an array of GeoJSON Features, returns an array of region details.
+ *
+ * Notes:
+ * - If a feature can be determined as having come from a known/supported BCGW layer, the region information from
+ * the feature will be added directly to the response.
+ * - If a feature can be determined as having come from a known/supported BCGW layer, then the geometry portion of the
+ * feature will be converted to Well-Known Text (WKT), and queried against the BCGW WFS service, returning matching
+ * region information, which will then be added to the response.
+ * - The response should not include duplicate information.
+ *
+ * @export
+ * @return {*}  {RequestHandler}
+ */
+export function getRegions(): RequestHandler {
+  return async (req, res) => {
+    const connection = getDBConnection(req['keycloak_token']);
+
+    try {
+      const features = req.body.features as Feature[];
+
+      let regionsDetails: RegionDetails[] = [];
+
+      const bcgwLayerService = new BcgwLayerService();
+
+      await connection.open();
+
+      for (const feature of features) {
+        const result = await bcgwLayerService.getRegionsForFeature(feature, connection);
+        regionsDetails = regionsDetails.concat(result);
+      }
+
+      await connection.commit();
+
+      // Convert array first into JSON, then into Set, then back to array in order to
+      // remove duplicate region information.
+      const regionDetailsJson = regionsDetails.map((value) => JSON.stringify(value));
```

Comment:
@NickPhura Had to change this to perform the `JSON.stringify` conversion, since a `Set` can only determine duplicates for primitives. Thoughts?

Reply:
Yeah I saw that, good catch on what I had before which was totally not working.

I think this works just fine atm. I hadn't really thought too much about it tbh. The only caveat would be if we imagine a situation where the object as a whole isn't valid for filtering duplicates (after adding some additional fields to it in future). Might never happen, but that is the only situation I can think of where this might break.

I found a stackoverflow post discussing this issue (https://stackoverflow.com/questions/2218999/how-to-remove-all-duplicates-from-an-array-of-objects), and the one upvoted solution seems to address that by just checking specific values in the object. I did try in a sandbox and it does work, not something i would have thought up on my own haha.

```
regionsDetails = regionsDetails.filter((value, index, self) =>
    index === self.findIndex((item) => (
        item.regionName === value.regionName && item.sourceLayer === value.sourceLayer
    ))
)
```